ktor-swagger/src/main/kotlin/de/nielsfalk/ktor/swagger/SwaggerSupport.kt

```diff
@@ -94,103 +198,82 @@ class SwaggerSupport(
             is BodyFromReflection -> bodyType.typeInfo.let { typeInfo ->
                 if (typeInfo.type != Unit::class) {
                     addDefinition(typeInfo)
                 }
             }
         }
 
-        fun createOperation(): Operation {
+        fun createOperation(): OperationBase {
             val responses = responses.map { (status, type) ->
                 val response = when (type) {
                     is ResponseFromReflection -> {
                         addDefinition(type.type)
-                        Response.create(status, type.type)
+                        variation.reponseCreator.create(status, type.type, type.examples)
                     }
                     is ResponseSchema -> {
-                        addDefintion(type.name, type.schema)
-                        Response.create(type.name)
+                        variation.reponseCreator.create(type.name, type.examples)
                     }
                 }
 
                 status.value.toString() to response
             }.toMap()
 
-            val parameters = mutableListOf<Parameter>().apply {
-                if ((bodyType as? BodyFromReflection)?.typeInfo?.type != Unit::class) {
-                    add(bodyType.bodyParameter())
-                }
-                addAll(locationType.memberProperties.map {
-                    it.toParameter(location.path).let {
-                        addDefinitions(it.second)
-                        it.first
+            val parameters = mutableListOf<ParameterBase>().apply {
+                variation {
+                    if ((bodyType as? BodyFromReflection)?.typeInfo?.type != Unit::class) {
+                        add(bodyType.bodyParameter())
                     }
-                })
-                parameter?.let {
-                    addAll(it.memberProperties.map {
-                        it.toParameter(location.path, ParameterInputType.query).let {
-                            addDefinitions(it.second)
-                            it.first
-                        }
-                    })
-                }
-                headers?.let {
-                    addAll(it.memberProperties.map {
-                        it.toParameter(location.path, ParameterInputType.header).let {
+                    addAll(locationType.memberProperties.map {
+                        it.toParameter(location.path).let {
                             addDefinitions(it.second)
                             it.first
                         }
                     })
+                    parameter?.let {
+                        addAll(it.memberProperties.map {
+                            it.toParameter(location.path, ParameterInputType.query).let {
+                                addDefinitions(it.second)
+                                it.first
+                            }
+                        })
+                    }
+                    headers?.let {
```

Comment:
Could some of this be put into a helper method? I'm seeing the same structure a few times here. It's hard to tell in the diff if it's repeated two or three times.